### ISSUE TITLE
Small-Medium User Data Analysis.

### ISSUE THREAD

**Issue by wmwv** [LSSTDESC/DC2_Repo#234]
DC2 Run 1.1 80GB;  66
        Run 1.2 1600 GB;  2000
        Run 2.0 16,000 TB; 20,000
coadd catalog

If we're talking 25 TB and 10 high-memory nodes.

1. Gateway.  Font-end node with smaller number of nodes behind it.  How much
2. Cloud

Fast-access implies a separate copy of the data connected to a set of fat nodes.
Start with ~100 TB and a dozen nodes to get a sense.  $50-100k you can do something.

Can IN2P3 support this?  Can we use something there?

Federated Identity/Access management?

Leadership Computing Facilities tend to be very security focused.  There may be value in setting something up somewhere with fewer concerns. 

NERSC is not necessarily a fan of support separate custom.

**Comment by wmwv** [comment]
Statement of Activity
Convert to resource estimate
Shop around.

**Comment by wmwv** [comment]
NERSC: MWV+Yao+Walter will try out Dask
IN2P3: Try out something in Spark
Argonne: Spark and related

25 TB is a department-level cluster.

**Comment by wmwv** [comment]
Put together a "DESC Box" to describe how to replicate.  Allow people to set things up.

Provides redundancy if we solve Federated Management.

**Comment by wmwv** [comment]
Ask LSST Project to host DC2 at NCSA as a full simulated dataset for useful studies, particularly for commissioning team, in exchange for DESC access.

**Comment by wmwv** [comment]
Consider NSF Gateway Project.

**Comment by wmwv** [comment]
AWS S3 Pricing:  https://aws.amazon.com/s3/pricing/
50 TB at $0.023/GB/month = $1,200/month

Google Cloud Storage:  https://cloud.google.com/storage/pricing
50 TB @ $0.026/GB/month = $1,330/month

**Comment by salmanhabib** [comment]
Better way to think about storage -- JBOD costs are $25K = 600TB (raw) or 400TB (RAID), so 100TB of RAIDed disk is roughly $6K, significantly cheaper than the cloud, which would cost about $30K/year (!). Actual cost ratio is roughly 20:1 given how long disks last.

Of course storage is not the only driver but this shows you what the (minimum) profit margins for Amazon and Google really are.

**Comment by salmanhabib** [comment]
Comment -- look into Globus solution for federated identity management (Globus Auth)?

**Comment by salmanhabib** [comment]
As discussed yesterday, one actionable thing to start with would be set up the minimal/desirable software stack (containerized) and an initial dataset that people can work with. We can plop this software+data object on certain available clusters -- at Argonne, BNL, commercial clouds, Fermilab, IN2P3, NCSA, NERSC, SLAC, various universities -- then based on early experience, figure out the next step in terms of the size and type of hardware investment.

**Comment by heather999** [comment]
Can we create a list of software that we want to include in such a minimal setup?  

- DMstack, including lsst_sims
- all DESC requested python packages as currently defined in the NERSC "desc-stack" environment
- What else?

**Comment by wmwv** [comment]
@heather999 Yes, for now, that list is fine.  Particularly in the spirit of getting started.

**Comment by wmwv** [comment]
@JulienPeloton Right now I'm looking at this through some Dask prototyping work (follow along on #237) on just doing a very simple aggregation on the difference between two columns (calculating the average color).  This would be a simple place to start and then we should think about some more sophisticated workflows and match science people want to do.

**Comment by JulienPeloton** [comment]
@wmwv Great thanks, this sounds like a good starting plan!
I'm taking holidays the next weeks, but I will report asap once back.

**Comment by JulienPeloton** [comment]
Question though. Looking at `/global/projecta/projectdirs/lsst/global/in2p3/Run1.1/object_catalog` at NERSC, I see `hdf5` files only. Is this the only option considered to store the data? No support for `fits` for example?

I'm asking because `hdf5` isn't (py)spark-friendly really... unless you are ready to work in Scala ;-)

**Comment by wmwv** [comment]
I'm happy to explore a variety of formats and it would be good to do so.  I picked `hdf5` because it was a standard supported by several data access methods.  Generating `FITS` is definitely doable.  Parquet is also something we'll be looking at in the near future.

**Comment by wmwv** [comment]
@JulienPeloton A full set of FITS, HDF5, and Parquet files have now been generated for Run 1.1p with column names following the DPDD.  There is one file per tract+path.  These files start with `dpdd_` on NERSC in:

`/global/projecta/projectdirs/lsst/global/in2p3/Run1.1/object_catalog`

I haven't even used these files for anything.  Please take a look and see if they are what you need.

**Comment by wmwv** [comment]
@JulienPeloton Do you have a sense of whether you would prefer per-tract or per-tract-patch FITS or Parquet files?

**Comment by JulienPeloton** [comment]
@wmwv Hum...  I'm not sure whether I can give a proper answer yet, but if we consider per-tract-patch FITS, will we end up with (1) less than a thousand, (2) a few thousands, or (3) more than hundreds thousand files?

For (1) and (2), per-tract or per-tract-patch FITS makes little difference for a Spark analysis.
For (3), there would be too many file system metadata involved, and we probably want to squeeze into per-tract FITS.

On aside, the current (per-tract-patch) is OK to me for the moment.

**Comment by JulienPeloton** [comment]
> Do you have a sense of whether you would prefer per-tract or per-tract-patch FITS or Parquet files?

more on this: 
- For Parquet hive scheme is just perfect (organised in dirs). It is super fast to load and simple to manipulate.
- For FITS, that depends what a per-tract file means. I understand this would be a single file per tract, with all patch data inside. But would that be 1 HDU per patch, or one big HDU with all patch data concatenated?

**Comment by wmwv** [comment]
That was going to be my follow-up question!

**Comment by JulienPeloton** [comment]
For FITS, the best would be one single big HDU with all patch data inside rather than several small ones. 
It would save a lot of metadata, and avoid unecessary IO.

**Comment by wmwv** [comment]
OK, I'll plan for that.

**Comment by katrinheitmann** [comment]
@wmwv This issue has not seen any action for more than 1.5 years. Should we close it? Should we do something about it? Thanks!
